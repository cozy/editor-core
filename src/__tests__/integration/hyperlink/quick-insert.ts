import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import {
  editable,
  getDocFromElement,
  fullpage,
  quickInsert,
  linkToolbar,
} from '../_helpers';

BrowserTestCase(
  'quick-insert.ts: Insert hyperlink via quick insert',
  { skip: ['edge', 'safari'] },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(fullpage.path);
    await browser.waitForSelector(editable);
    await browser.click(editable);
    await quickInsert(browser, 'Link');

    await browser.waitForSelector(linkToolbar);
    await browser.type(linkToolbar, ['google.com']);
    await browser.keys(['Return']);
    await browser.waitForSelector('a');

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
