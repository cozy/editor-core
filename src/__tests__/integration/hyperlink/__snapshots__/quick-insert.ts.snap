// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quick-insert.ts: Insert hyperlink via quick insert 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://google.com",
              },
              "type": "link",
            },
          ],
          "text": "google.com",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
